src/ModelContextProtocol/Protocol/Transport/StdioClientTransport.cs

```diff
@@ -1,31 +1,17 @@
 ﻿using Microsoft.Extensions.Logging;
-using Microsoft.Extensions.Logging.Abstractions;
 using ModelContextProtocol.Configuration;
-using ModelContextProtocol.Logging;
-using ModelContextProtocol.Protocol.Messages;
 using ModelContextProtocol.Utils;
-using ModelContextProtocol.Utils.Json;
-using System.Diagnostics;
-using System.Text;
-using System.Text.Json;
 
 namespace ModelContextProtocol.Protocol.Transport;
 
 /// <summary>
 /// Implements the MCP transport protocol over standard input/output streams.
 /// </summary>
-public sealed class StdioClientTransport : TransportBase, IClientTransport
+public sealed class StdioClientTransport : IClientTransport
```

Comment:
Oh, I see, the public StdioClientTransport wraps an internal StdioClientStreamTransport.

That's fine if it's useful, but what's the rationale for that factoring? This type seems like the only consumer.

Reply:
The rational is that the McpClient creates the stdio session ITransport from connecting using the IClientTransport. This makes it the owner of the ITransport and the responsibility for disposing it and ending the session. This also means that the IClientTransport can be reused since the McpClient doesn't dispose that.

Reply:
I'm having trouble following. It seems like you're saying there are multiple layers of transports, one for the session and one for something else?

Reply:
The outer layer is useful for establishing a new session and clarifying ownership of the session. Whoever connects/accepts is the owner. It also means that whenever you get an ITransport, it's already connected. It's now impossible to get an ITransport in a preconnected state. The IServerTransport and IClientTransport are almost identical aside from the names nullability of the ITransport returned. The server one is nullable, so the host knows to exit when AcceptAsync returns null.

In the case of the StdioServerTransport in particular, the ability to establish multiple session in a single process isn't that useful, and consumers were already accustomed to being able to get an IMcpServer instance synchronously, so I left that as a plain single-session ITransport.